src/BackupMetadata.cs

```diff
@@ -43,6 +29,28 @@ public int GetHashCode(BackupMetadata obj)
       hashCode = hashCode * -1521134295 + obj.DatabaseBackupLsn.GetHashCode();
       return hashCode;
     }
+
+  }
+
+  /// <summary>
+  /// Two BackupMetadatas are the same, if they are like striped backups but also have the same `PhysicalDeviceName`
+  /// </summary>
+  public class BackupMetadataEqualityComparer : IEqualityComparer<BackupMetadata>
+  {
+    private readonly StripedBackupEqualityComparer _stripedBackupEqualityComparer = new StripedBackupEqualityComparer();

```

Comment:
I might suggest that you look into creating a static, singleton, instance of this class.
You call `new StripedBackupEqualityComparer` several times that seem unnecessary.  I think this would all work with 1 instance of the class. Let me know

Reply:
Agreed! Made them a singleton